enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -1751,18 +1752,39 @@ class AdjointGenerator
     }
   }
 
-  void subTransferHelper(Type *secretty, BasicBlock *parent,
-                         Intrinsic::ID intrinsic, unsigned dstalign,
-                         unsigned srcalign, unsigned offset, Value *orig_dst,
-                         Value *orig_src, Value *length, Value *isVolatile,
-                         llvm::CallInst *MTI, bool allowForward = true) {
+  void subTransferHelper(Type *secretty, llvm::MemTransferInst &MTI,
+                         unsigned dstalign, unsigned srcalign, unsigned offset,
+                         Value *length, bool allowForward = true) {
+
+    BasicBlock *parent = MTI.getParent();
```

Comment:
I don't think you want to call this for forward mode (it is designed to do the += required by reverse, with correctness for types). You should simply be able to do a memcpy regardless of type, and since you only need to set not increment, this shouldn't be required?